actors/sealer/src/lib.rs

```diff
@@ -0,0 +1,174 @@
+use crate::ext::account;
+pub use self::state::{State, CollisionPolicy};
+
+pub mod ext;
+mod state;
+pub mod types;
+pub mod testing;
+
+#[cfg(feature = "fil-actor")]
+fil_actors_runtime::wasm_trampoline!(Actor);
+
+/// Sealer actor methods available
+#[derive(FromPrimitive)]
+#[repr(u64)]
+pub enum Method {
+    Constructor = METHOD_CONSTRUCTOR,
+    
+    ActivateSectors = frc42_dispatch::method_hash!("ActivateSectors"),
+    CompactSectorNumbers = frc42_dispatch::method_hash!("CompactSectorNumbers"),
+}
+
+/// Sealer Actor
+pub struct Actor;
+
+impl Actor {
+    /// Constructor for Sealer actor
+    pub fn constructor(rt: &impl Runtime, _params: ConstructorParams) -> Result<(), ActorError> {
+        rt.validate_immediate_caller_is(std::iter::once(&INIT_ACTOR_ADDR))?;
+
+        let empty_bitfield = rt.store().put_cbor(&BitField::new(), Code::Blake2b256).map_err(|e| {
+            e.downcast_default(ExitCode::USR_ILLEGAL_STATE, "failed to construct empty bitfield")
+        })?;
+
+        let validator = _params.validator;
+        let state = State {
+            validator,
+            allocated_sectors: empty_bitfield,
+        };
+        rt.create(&state)?;
+        Ok(())
+    }
+
+    pub fn activate_sectors(rt: &impl Runtime, params: ActivateSectorParams) -> Result<ActivateSectorReturn, ActorError> {
+        rt.validate_immediate_caller_type(std::iter::once(&Type::Miner))?;
+
+        rt.transaction(|state: &mut State, rt| {
+
+            // Call the validator with the sector numbers
+            let payload = types::VerifierSignaturePayload::new(
+                params.sector_numbers.clone(),
+                rt.message().receiver(),
+                rt.message().caller(),
+            );
+            
+            let serialized_payload = payload.serialize()
+                .map_err(|e| actor_error!(illegal_state, "failed to serialize payload: {}", e))?;
+            
+            // We're not actually signing anything here, just passing the payload to the validator
+            // The validator will verify the sector numbers are valid
+            let auth_params = account::AuthenticateMessageParams {
+                signature: params.verifier_signature,
+                message: serialized_payload,
+            };
+            
+            // Call the validator actor to authenticate the sector numbers
+            let send_flags = SendFlags::default();
+
+            rt.send(
+                &state.validator,
+                account::AUTHENTICATE_MESSAGE_METHOD,
+                IpldBlock::serialize_cbor(&auth_params)?,
+                TokenAmount::zero(),
+                None,
+                send_flags,
+            )?;
+            
+            // Allocate the sector numbers after validation
+            state.allocate_sector_numbers(
+                rt.store(),
+                &params.sector_numbers,
+                CollisionPolicy::DenyCollisions,
```

Comment:
Based on this it looks like we might have a different idea of the security concern of collisions.  I think we shouldn't even configure this and always deny collisions as a matter of network security.  I want to hear your thoughts though -- it looks like none of this was really specified in the draft FIP.

Reply:
This is matching the same exact logic in the miner actor, the allocate_sector_numbers is pretty much copy-paste of the miner one

Reply:
Ok that makes sense then

Reply:
I would either remove this over engineered collision policy thing in this copied code or extract the allocation function somewhere so miner and sealer can share it.